### PR TITLE
Fix connection validator user privilege check

### PR DESCRIPTION
**Pull request by rozza**
Now uses connection status to determine the users privileges.
Handles authentication mechanisms that don't require a username.

KAFKA-238

### CODE REVIEW

**Comment by rozza** [comment]
Changing to draft - the build is failing - investigating

File: src/integrationTest/java/com/mongodb/kafka/connect/ConnectorValidationIntegrationTest.java [hunk 1]
```diff
@@ -474,7 +474,14 @@ private void assertInvalidSource(final Map<String, String> properties) {
   private void assertInvalidSource(final Map<String, String> properties, final String configName) {
     Optional<ConfigValue> configValue =
         getSourceErrors(properties).stream().filter(cv -> cv.name().equals(configName)).findFirst();
-    assertTrue(configValue.isPresent());
+    assertTrue(
```
Comment (rozza) [inline src/integrationTest/java/com/mongodb/kafka/connect/ConnectorValidationIntegrationTest.java:477]: Improve debugging

File: src/main/java/com/mongodb/kafka/connect/util/ConnectionValidator.java [hunk 2]
```diff
@@ -182,26 +168,65 @@ public static void validateUserHasActions(
...
+          .ifPresent(
+              c ->
+                  c.addErrorMessage(
+                      "Invalid user permissions authentication failed. " + e.getMessage()));
     } catch (Exception e) {
       LOGGER.warn("Permission validation failed due to: {}", e.getMessage(), e);
     }
   }
 
+  private static List<String> validateRoleHasActions(
```
Comment (stIncMale) [inline src/main/java/com/mongodb/kafka/connect/util/ConnectionValidator.java:180]: This method behaves similarly to `removeUserActions`: it gets a set of actions and returns a subset representing those actions that are not permitted. However, the naming is different, which negatively contributes to readability.

Another thing I noticed is that the method `removeUserActions` does not modify the passed `actions` collection, which yet its name implies modification.

Can we name this methods in a more consistent and descriptive way?

**Review by stIncMale — CHANGES_REQUESTED** [review]

File: src/integrationTest/java/com/mongodb/kafka/connect/ConnectorValidationIntegrationTest.java [hunk 3]
```diff
@@ -474,7 +474,14 @@ private void assertInvalidSource(final Map<String, String> properties) {
...
+            configValue.isPresent(),
+            format(
+                    "No error for '%s': %s\nErrors: %s\nProperties: %s",
```
[the remark this answers is not in the document]
Comment (rozza) [reply src/integrationTest/java/com/mongodb/kafka/connect/ConnectorValidationIntegrationTest.java:480]: `assertInvalidSource` expects there to be an error for the provided configuration.  The errors are filtered so if the configValue isn't present there has been no error.

File: src/main/java/com/mongodb/kafka/connect/util/ConnectionValidator.java (hunk 2 above)
Comment (rozza) [reply src/main/java/com/mongodb/kafka/connect/util/ConnectionValidator.java:180]: Will do

File: src/integrationTest/java/com/mongodb/kafka/connect/ConnectorValidationIntegrationTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (stIncMale) [reply src/integrationTest/java/com/mongodb/kafka/connect/ConnectorValidationIntegrationTest.java:480]: Thank you, I have now understood that the `Optional<ConfigValue>` is supposed to contain not a valid `ConfigValue`, but a `ConfigValue` that caused an error.

**Review by stIncMale — APPROVED** [review]

**Pull request merged**